src/pdn/src/straps.cpp

```diff
@@ -616,6 +616,11 @@ void PadDirectConnectionStraps::initialize(ConnectionType type)
       break;
   }
 
+  std::sort(pins_.begin(), pins_.end(), [](odb::dbBox* l, odb::dbBox* r) {
+    return std::make_tuple(l->xMin(), l->xMax(), l->yMin(), l->yMax())
+           < std::make_tuple(r->xMin(), r->xMax(), r->yMin(), r->yMax());
+  });
```

Comment:
While sorting here does close the discrepancy between my local runs and the CI, @gadfort points out `pins_` should be having a stable order by construction. The root cause for test flakiness might lie elsewhere

Reply:
I actually think a much more likely culprit might be the rtree storing the shapes. This would cause the order in which we write the shapes to the database to differ because the order in the tree is different.